packages/Operators/test/tstNodeToNodeOperator.cpp

```diff
@@ -0,0 +1,196 @@
+    Teuchos::Array<DataTransferKit::Entity> range_points( num_points );
+    test_result.resize( field_dim*num_points );
+    gold_data.resize( num_points );
+    for ( int i = 0; i < num_points; ++i )
+    {
+	point_id = num_points*inverse_rank + i + 1;
+        coord_val = static_cast<double>(i) / num_points;        
+	coords[0] = coord_val + inverse_rank;
+	coords[1] = coord_val;
+	coords[2] = coord_val;
+	range_points[i] = DataTransferKit::Point( point_id, comm_rank, coords );
+	test_result[i] = 0.0;
+	gold_data[i] = coords[0] + coords[1] + coords[2];
+    }
+
+    // Make a manager for the domain geometry.
+    DataTransferKit::BasicGeometryManager domain_manager( 
+	comm, space_dim, domain_points() );
+					   
+    // Make a manager for the range geometry.
+    DataTransferKit::BasicGeometryManager range_manager( 
+	comm, space_dim, range_points() );
+
+    // Make a DOF vector for the domain.
+    Teuchos::RCP<DataTransferKit::Field> domain_field =
+	Teuchos::rcp( new DataTransferKit::EntityCenteredField(
+			  domain_points(), field_dim, domain_data,
+			  DataTransferKit::EntityCenteredField::BLOCKED) );
+    Teuchos::RCP<Tpetra::MultiVector<double,int,DataTransferKit::SupportId> > domain_vector =
+	Teuchos::rcp( new DataTransferKit::FieldMultiVector(
+			  domain_field,
+			  domain_manager.functionSpace()->entitySet()) );
+
+    // Make a DOF vector for the range.
+    Teuchos::RCP<DataTransferKit::Field> range_field =
+	Teuchos::rcp( new DataTransferKit::EntityCenteredField(
+			  range_points(), field_dim, Teuchos::arcpFromArray(test_result),
+			  DataTransferKit::EntityCenteredField::BLOCKED) );
+    Teuchos::RCP<Tpetra::MultiVector<double,int,DataTransferKit::SupportId> > range_vector =
+	Teuchos::rcp( new DataTransferKit::FieldMultiVector(
+			  range_field,
+			  range_manager.functionSpace()->entitySet()) );
+
+    // Create the point cloud operator
+    DataTransferKit::MapOperatorFactory factory;
+    Teuchos::RCP<DataTransferKit::MapOperator> cloud_op =
+	factory.create( domain_vector->getMap(), range_vector->getMap(), *parameters );
+
+    // Setup the operator.
+    cloud_op->setup( domain_manager.functionSpace(),
+		     range_manager.functionSpace() );
+
+    // Apply the operator.
+    cloud_op->apply( *domain_vector, *range_vector );
+}
+
+//---------------------------------------------------------------------------//
+// Tests.
+//---------------------------------------------------------------------------//
+TEUCHOS_UNIT_TEST( NodeToNodeOperator, node_to_node_test )
+{
+    // Run the test.
+    Teuchos::Array<double> gold_data;
+    Teuchos::Array<double> test_result;
+    setupAndRunTest(
+	"node_to_node_test.xml", gold_data, test_result );
+    
+    // Check the results.
+    TEST_EQUALITY( gold_data.size(), test_result.size() );
+    int num_points = gold_data.size();
+    for ( int i = 0; i < num_points; ++i )
+    {
+	TEST_FLOATING_EQUALITY( gold_data[i], test_result[i], epsilon );
+    }
```

Comment:
did you check or ask whether these unit testing macros from Teuchos can be called inside a function?